cairo/ethereum/cancun/fork.cairo

```diff
@@ -855,3 +885,250 @@ func keccak256_header{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr:
+            block_gas_used=block_gas_used,
+            transactions_root=dummy_root,
+            receipt_root=dummy_root,
+            block_logs_bloom=block_logs_bloom,
+            state_root=dummy_root,
+            withdrawals_root=dummy_root,
+            blob_gas_used=blob_gas_used,
+        ),
+    );
+    return output;
+}
+
+func _apply_body_inner{
+    range_check_ptr,
+    range_check96_ptr: felt*,
+    add_mod_ptr: ModBuiltin*,
+    mul_mod_ptr: ModBuiltin*,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    state: State,
+}(
+    index: felt,
+    len: felt,
+    transactions: TupleUnionBytesLegacyTransaction,
+    gas_available: Uint,
+    chain_id: U64,
+    base_fee_per_gas: Uint,
+    excess_blob_gas: U64,
+    block_logs: TupleLog,
+    block_hashes: ListHash32,
+    coinbase: Address,
+    block_number: Uint,
+    block_gas_limit: Uint,
+    block_time: U256,
+    prev_randao: Bytes32,
+    blob_gas_used: Uint,
+) -> (blob_gas_used: Uint, gas_available: Uint, block_logs: TupleLog) {
+    alloc_locals;
+    if (index == len) {
+        return (blob_gas_used, gas_available, block_logs);
+    }
+
+    let tx = decode_transaction(transactions.value.data[index]);
+
+    // TODO: call transactions_trie_set
+    // trie_set(
+    //         transactions_trie, rlp.encode(Uint(i)), encode_transaction(tx)
+    //     )
+
+    let tuple_address_uint_tuple_versioned_hash = check_transaction{state=state}(
+        tx, gas_available, chain_id, base_fee_per_gas, excess_blob_gas
+    );
+    let sender_address = tuple_address_uint_tuple_versioned_hash.value.address;
+    let effective_gas_price = tuple_address_uint_tuple_versioned_hash.value.uint;
+    let blob_versioned_hashes = tuple_address_uint_tuple_versioned_hash.value.tuple_versioned_hash;
+
+    let transient_storage = empty_transient_storage();
+
+    tempvar env = Environment(
+        new EnvironmentStruct(
+            caller=sender_address,
+            block_hashes=block_hashes,
+            origin=sender_address,
+            coinbase=coinbase,
+            number=block_number,
+            base_fee_per_gas=base_fee_per_gas,
+            gas_limit=block_gas_limit,
+            gas_price=effective_gas_price,
+            time=block_time,
+            prev_randao=prev_randao,
+            state=state,
+            chain_id=chain_id,
+            excess_blob_gas=excess_blob_gas,
+            blob_versioned_hashes=blob_versioned_hashes,
+            transient_storage=transient_storage,
+        ),
+    );
+
+    let (gas_used, tuple_logs, optional_error) = process_transaction{env=env}(tx);
+    let state = env.value.state;
+
+    // Safe because gas_used <= gas_available
+    tempvar gas_available = Uint(gas_available.value - gas_used.value);
+
+    tempvar receipt_gas = Uint(block_gas_limit.value - gas_available.value);
+    let receipt = make_receipt(tx, optional_error, receipt_gas, tuple_logs);
```

Comment:
```suggestion
    let (gas_used, tuple_logs, error) = process_transaction{env=env}(tx);
    let state = env.value.state;

    // Safe because gas_used <= gas_available
    tempvar gas_available = Uint(gas_available.value - gas_used.value);

    tempvar receipt_gas = Uint(block_gas_limit.value - gas_available.value);
    let receipt = make_receipt(tx, error, receipt_gas, tuple_logs);
```